client/src/app/utils/utils.test.ts

```diff
@@ -89,4 +93,20 @@ describe("utils", () => {
     const result = getToolbarChipKey({ key: "myKey", node: "myNode" });
     expect(result).toBe("myKey");
   });
+
+  //URL Regex tests
+  it("Regex should validate repo URLs & git URLs", () => {
+    const gitURLExample = "git@github.com:konveyor/tackle2-ui";
+    const gitValidationResult = gitUrlRegex.test(gitURLExample);
+    expect(gitValidationResult).toBe(true);
+
+    const customRuleRepoURLExample =
+      "https://github.com/ibolton336/tackle-testapp.git";
+    const URLvalidationResult = gitUrlRegex.test(customRuleRepoURLExample);
+    expect(URLvalidationResult).toBe(true);
+
+    const brokenURLExample = "github.com/ibolton336/tackle-testapp.git";
+    const brokenURLvalidationResult = gitUrlRegex.test(brokenURLExample);
+    expect(brokenURLvalidationResult).toBe(false);
+  });
```

Comment:
I suggest adding a few more strings to the tests, and have an `it()` section for `gitUrlRegex` and `standardURLRegex`.

A few url string suggestions:
  - `""`
  - `"     http://www.foo.bar"`
  - `"     http://www.foo.bar     "`
  - `"www.foo.bar"`
  - `"foo.bar"`
  - `"www.foo.b"`
  - `"foo.ba"`

Having a separate `it()` just for a double test would be good as well:
```js
it("URL should match the same multiple times in a row", () => {
  const url = "https://github.com/ibolton336/tackle-testapp.git";

  expect(standardURLRegex.test(url).toBe(true);
  expect(standardURLRegex.test(url).toBe(true);
  expect(standardURLRegex.test(url).toBe(true);
  expect(standardURLRegex.test(url).toBe(true);
  expect(standardURLRegex.test(url).toBe(true);
});
```

And that test should fail if you put the `/g` back on the pattern...

Reply:
jeez, I would not have expected a regex `test` to have state like that